libraries/AP_RCProtocol/AP_RCProtocol.cpp

```diff
@@ -118,19 +112,29 @@ void AP_RCProtocol::process_pulse(uint32_t width_s0, uint32_t width_s1)
         !protocol_enabled(_detected_protocol)) {
         _detected_protocol = AP_RCProtocol::NONE;
     }
-    
+
     if (_detected_protocol != AP_RCProtocol::NONE && _detected_with_bytes && !searching) {
         // we're using byte inputs, discard pulses
         return;
     }
     // first try current protocol
+    uint8_t b;  // SoftSerial is stateful so only get the current byte once
     if (_detected_protocol != AP_RCProtocol::NONE && !searching) {
-        backend[_detected_protocol]->process_pulse(width_s0, width_s1);
+        if (!backend[_detected_protocol]->process_pulse(width_s0, width_s1)) {
+            if (added.ss.process_pulse(width_s0, width_s1, b)) {
+                backend[_detected_protocol]->process_byte(added.ss.get_byte_timestamp_us(), b, added.ss.baud());
+            }
+        }
+
         if (backend[_detected_protocol]->new_input()) {
             _new_input = true;
-            _last_input_ms = now;
+            _last_input_us = now;
         }
         return;
+    } else {
+        if (!added.ss.process_pulse(width_s0, width_s1, b)) {
```

Comment:
This change means process pulse never gets called for non serial protocols during searching step. PPM for instance will stop working.